frontend/app/components/Spots/SpotsList/SpotListItem.tsx

```diff
@@ -86,49 +91,86 @@ function SpotListItem({ spot, onRename, onDelete, onVideo, onSelect }: ISpotList
   };
 
+  const copyToClipboard = () => {
+    const spotLink = withSiteId(spotUrl(spot.spotId.toString()), siteId);
+    const fullLink = `${window.location.origin}${spotLink}`;
+    navigator.clipboard.writeText(fullLink)
+      .then(() => {
+        setTooltipText('Link copied to clipboard!');
+        setTimeout(() => setTooltipText('Copy link to clipboard'), 2000); // Reset tooltip text after 2 seconds
+      })
+      .catch(() => {
+        setTooltipText('Failed to copy URL');
+        setTimeout(() => setTooltipText('Copy link to clipboard'), 2000); // Reset tooltip text after 2 seconds
+      });
+  };
+
   const onSave = (newName: string) => {
     onRename(spot.spotId, newName);
     setIsEdit(false);
-  }
+  };
+
   return (
     <div
       className={
-        'border rounded-xl overflow-hidden flex flex-col items-start hover:shadow'
+        'bg-white rounded-lg overflow-hidden shadow-sm border border-transparent transition flex flex-col items-start hover:border hover:border-teal'
       }
     >
       {isEdit ? (
         <EditItemModal onSave={onSave} onClose={() => setIsEdit(false)} itemName={spot.title} />
       ) : null}
-      <div style={{ cursor: 'pointer', width: '100%', height: 180, position: 'relative' }} onClick={onSpotClick}>
-        <img
-          src={spot.thumbnail}
-          alt={spot.title}
-          className={'w-full h-full object-cover'}
-        />
+      <div className='relative group overflow-hidden bg-black' style={{ width: '100%', height: 180 }}>
+        {loading && (
+          <div className="absolute inset-0 flex items-center justify-center">
+            <AnimatedSVG name={ICONS.LOADER} size={32} />
+          </div>
+        )}
+        <div className='block w-full h-full cursor-pointer transition  hover:bg-teal/50' onClick={onSpotClick}>
+          <img
+            src={spot.thumbnail}
+            alt={spot.title}
+            className={'w-full h-full object-cover opacity-80'}
+            onLoad={() => setLoading(false)}
+            onError={() => setLoading(false)}
+            style={{ display: loading ? 'none' : 'block' }}
+          />
+        </div>
+
+        <div className='absolute left-0 bottom-8 flex relative gap-2 justify-end pe-2 pb-2 transition-transform transform translate-y-full group-hover:translate-y-0 group-hover:opacity-100 opacity-0'>
+        
+        <Tooltip title={tooltipText}>
+            <div
+              className={
+                ' bg-black/70 text-white p-1 px-2 text-xs rounded-lg text-white'
+              }
+              onClick={copyToClipboard}
+              style={{ cursor: 'pointer' }}
+            >
+              <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" className="lucide lucide-link-2"><path d="M9 17H7A5 5 0 0 1 7 7h2"/><path d="M15 7h2a5 5 0 1 1 0 10h-2"/><line x1="8" x2="16" y1="12" y2="12"/></svg>
```

Comment:
We don't include svgs in react codebase. Drop it in icons folder, run `npm run gen:icons` and use it as part of <Icon> UI component please

Reply:
Used Lucide icon in the updated version.